### PR TITLE
doc: guide for static PVC creation

### PR DESCRIPTION
**Pull request by MondayCha**
# Describe what this PR does #

Add a detailed guide on creating a static PVC for a CephFS volume.

## Related discussions ##

Discussions: #4906

### CODE REVIEW

File: docs/static-pvc.md [hunk 1]
```diff
@@ -215,6 +225,45 @@ spec:
   volumeMode: Filesystem
 ```
 
+If you are a [Rook](https://rook.io/) user, there are some differences, such as:
```
Comment (iPraveenParihar) [inline docs/static-pvc.md]: The above example should be enough. Let's not have Rook specific example here and anyway there isn't any extra configuration needed for Rook.

File: docs/static-pvc.md [hunk 2]
```diff
@@ -269,5 +279,42 @@ $ kubectl create -f cephfs-static-pvc.yaml
 persistentvolumeclaim/cephfs-static-pvc created
 ```
 
-**Note** deleting PV and PVC does not delete the backend CephFS subvolume,
+### Verify PVC creation
```
Comment (Madhu-1) [inline docs/static-pvc.md]: can you please update the rbd example with the verification step as well?

Comment (MondayCha) [reply docs/static-pvc.md]: Sure, but in our scenario, we use OpenEBS to provide LocalPV for database service persistence, so I cannot provide a solution immediately. 

I will test it in the cluster and provide a verification method suitable for RBD🫡.

File: docs/static-pvc.md [hunk 3]
```diff
@@ -140,24 +145,69 @@ size to match the size of the rbd image.
...
+pod/rbd-static-pvc-test created
+```
+
+Verify that the PVC has been successfully mounted.
+
+```bash
+$ kubectl exec rbd-static-pvc-test -- df -h /data/pvc
+```
+
+After completing the verification steps, you can delete the test Pods for RBD:
```
Comment (Madhu-1) [inline docs/static-pvc.md]:
```suggestion
Once you have completed the verification step, you can delete the test Pod for RBD.
```

File: docs/static-pvc.md [hunk 4]
```diff
@@ -269,5 +322,49 @@ $ kubectl create -f cephfs-static-pvc.yaml
 persistentvolumeclaim/cephfs-static-pvc created
 ```
 
-**Note** deleting PV and PVC does not delete the backend CephFS subvolume,
-user needs to manually delete the CephFS subvolume if required.
+### Verify CephFS static PVC creation
```
Comment (Madhu-1) [inline docs/static-pvc.md]:
```suggestion
### Verify CephFS static PVC
```

File: docs/static-pvc.md [hunk 5]
```diff
@@ -269,5 +322,49 @@ $ kubectl create -f cephfs-static-pvc.yaml
...
+
+We can verify the CephFS static PVC creation by running a Pod that mounts the PVC.
```
Comment (Madhu-1) [inline docs/static-pvc.md]:
```suggestion
We can verify the CephFS static PVC by creating a Pod that mounts the pvc.
```

File: docs/static-pvc.md [hunk 6]
```diff
@@ -140,24 +145,69 @@ size to match the size of the rbd image.
...
-**Note** If you have mounted same static PVC to multiple application pods, make
+> [!note]
+> If you have mounted same static PVC to multiple application pods, make
 sure you will scale down all the application pods and make sure no application
 pods using the static PVC is running on the node and scale up all the
 application pods again(this will trigger `NodeStageVolumeRequest` which will
 resize the filesystem for static volume).
 
-**Note** deleting PV and PVC does not removed the backend rbd image, user need to
+### Verify RBD static PVC creation
```
Comment (Madhu-1) [inline docs/static-pvc.md]:
```suggestion
### Verify RBD static PVC
```

File: docs/static-pvc.md [hunk 7]
```diff
@@ -140,24 +145,69 @@ size to match the size of the rbd image.
...
+
+We can verify the RBD static PVC creation by running a Pod that mounts the PVC.
```
Comment (Madhu-1) [inline docs/static-pvc.md]:
```suggestion
We can verify the RBD static PVC by creating a Pod that mounts the pvc.
```

**Review by Madhu-1 — COMMENTED** [review]
small nits.

**Comment by MondayCha** [comment]
> small nits.

I apologize for my carelessness 😢.

**Comment by Madhu-1** [comment]
@MondayCha can you please take care of the lint failure in lint-extra?

**Comment by nixpanic** [comment]
The **multi-arch-build** CI job will be fixed when this PR is rebased.

**Review by Madhu-1 — APPROVED** [review]

**Review by iPraveenParihar — APPROVED** [review]
LGTM. Thanks!

**Comment by nixpanic** [comment]
@mergifyio queue

**Comment by MondayCha** [comment]
@mergifyio requeue

**Comment by MondayCha** [comment]
I tried to check the reason for the failure, but the information showed:

> [!warning]
> The merge queue pull request cannot be updated.
>
>Details:
>
> It is unable to update: The user MondayCha is unknown.
>
>Please make sure that MondayCha has logged in to the Mergify dashboard.

I read the [development documentation](https://github.com/ceph/ceph-csi/blob/devel/docs/development-guide.md) again, but it didn't mention the need to register with Mergify. I also tried to log in to Mergify, but it seems that user information can't be viewed on it. 

Could anyone teach me how to do it?

**Comment by nixpanic** [comment]
@mergifyio rebase

**Comment by nixpanic** [comment]
@mergifyio queue

**Pull request merged**